_docs/geyser/extensions.md

```diff
@@ -0,0 +1,123 @@
+There is an [official list](https://github.com/GeyserMC/GeyserExtensionList) of available extensions you can check out.
+
+### Installing Extensions
+To install an extension, simply put the extension .jar file into Geyser's 'extensions' folder. Then, restart Geyser (or the server Geyser runs on).
+
+### Creating Geyser Extensions
+The easiest way to create an extension would be utilizing [this official template](https://github.com/GeyserMC/GeyserExampleExtension/). Simply create a new repository from the template, customize the 'extension.yml' & 'settings.gradle' files, and get started making the extension.
+
+Geyser recognizes extensions when they have a file called 'extension.yml' in the jars 'resources' folder.
+
+extension.yml:
+```yml
+id: exampleid
+name: ExampleExtension
+main: org.geyser.extension.exampleid.ExampleExtension
+api: 2.1.1
+version: 1.0.0
+authors: [ExampleAuthor]
+```
+
+Explanations for the individual fields:
+- id: The id of the extension. Every extension needs to have their unique id - all lowercase letters. If you e.g. register a command for your extension, it will have the id as a prefix: e.g. '/exampleid command'.
+- name: The name of the extension.
+- main: The main class of your extension.
+- api: The Geyser api version your extension targets.
+- authors: The author(s) of the extension. To add more than one entry, separate entries with a comma.
+
+### Creating the main class
+
+The main class, the entrypoint for the extension, needs to [implement the 'Extension' interface provided by Geyser](https://github.com/GeyserMC/GeyserExampleExtension/blob/47614575a69bddecb241676215f3c9f9113db304/src/main/java/org/geyser/extension/exampleid/ExampleExtension.java#L10). 
+That way, Geyser recognizes the extension, and gives you access to important methods - such as 'logger()', to get your extensions logger. <br>
+To see all the methods provided by that interface, see [here](https://github.com/GeyserMC/Geyser/blob/master/api/src/main/java/org/geysermc/geyser/api/extension/Extension.java).
+
+Unlike plugins, extensions do not have a 'onEnable' or 'onDisable' method. Instead, most actions are done in events at different stages during Geyser's lifecycle.
+Some important ones are:
+- `GeyserPreInitializeEvent`: This event is fired when Geyser starts to initialize. If you e.g. need to register extension commands that are configured in your config, 
+you would need to load the config here to ensure that your config is ready before the GeyserDefineCommandsEvent is fired. 
+- `GeyserPostInitializeEvent`: It is called when Geyser has completed initializing. The bulk of your code should go here, as the GeyserAPI is fully available at this stage.
+- `GeyserShutdownEvent`: Called when Geyser is shutting down. You can use this to e.g. save data, or clean up resources.
+
+The bulk of your code would go into the `GeyserPostInitializeEvent`, when Geyser is ready to accept & handle players. See below for an example:
```

Comment:
The fact that the bulk of code should be here is mentioned earlier.

Reply:
fixed; ty